### PR TITLE
Fix issue with `no-deprecated-router-transition-methods` throwing errors outside of class usage

### PR DESCRIPTION
**Pull request by NullVoxPopuli**
Ran in to this while trying out eslint-plugin-ember@v12-alpha on https://github.com/NullVoxPopuli/limber/pull/1604

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -73,7 +73,6 @@
   },
   "dependencies": {
     "@ember-data/rfc395-data": "^0.0.4",
-    "@glimmer/syntax": "^0.86.0",
```
Comment (NullVoxPopuli) [inline package.json:76]: this was unused and was meant to be pulled out earlier -- it _is_ used by the parser tho

Comment (bmish) [reply package.json:76]: Would be best to fix in separate PR.

Comment (NullVoxPopuli) [reply package.json]: done, https://github.com/ember-cli/eslint-plugin-ember/pull/2047

**Review by bmish — APPROVED** [review]
Thanks!

**Pull request merged**